blockstore/splitstore/splitstore.go

```diff
@@ -1635,6 +1702,8 @@ func (s *SplitStore) waitForMissingRefs(markSet MarkSet) {
 		}
 	}
 
+	// Do we really want to proceed here? Maybe we should just abort and try again later?
+	// Also, aren't these errors?
```

Comment:
Also, someone syncing off us and reading old headers will trigger this as well; this vector will go away once #6705 is done though.

Reply:
> TL;DR They don't seem to be errors at all and there is a legitimate scenario where they can arise: a lookback for randomness references an old block header whose messages and state root have been garbage collected (or were never there in the first place because we synced from snapshot).

Ah, I see. Because we're keeping everything we've touched. We _really_ need VM integration.

Reply:
yeah, agreed.